mlir/lib/Dialect/Vector/IR/VectorOps.cpp

```diff
@@ -5202,11 +5219,28 @@ class GatherFolder final : public OpRewritePattern<GatherOp> {
     llvm_unreachable("Unexpected 1DMaskFormat on GatherFolder");
   }
 };
+
+/// Fold gathers with consecutive offsets [0, 1, 2, ...] into contiguous
+/// maskedload. Only 1D non-scalable vectors are supported for now.
```

Comment:
Please add tests for scalable vectors and highlight where in the implementation the assumption about "non-scalability" is made.  Also, please add a negative test with 2D vectors.

[nit] We tend to say "fixed width vs scalable" rather than "non-scalable vs scalable" :_